File: src/frontend/src/app/components/atendimento/agendamento/agendamento/model/agendamento-dto.ts
import { Exclude, Expose } from 'class-transformer';

@Exclude()
export class AgendamentoDTO {
  @Expose() id?: string;
  @Expose() agendaId?: string;
  @Expose() agendaNome?: string;
  @Expose() profissionalId?: string;
  @Expose() profissionalNome?: string;
  @Expose() pacienteId?: string;
  @Expose() pacienteNome?: string;
  @Expose() convenioId?: string;
  @Expose() convenioNome?: string;
  @Expose() categoriaId?: string;
  @Expose() categoriaNome?: string;
  @Expose() procedimentoId?: string;
  @Expose() procedimentoNome?: string;
  @Expose() observacao?: string;
  @Expose() status?: string;
  @Expose() atendimentoId?: string;
  @Expose() horariosInicio?: string[];
  @Expose() horariosFim?: string[];
  @Expose() createdAt?: string;
  @Expose() updatedAt?: string;
  @Expose() createdBy?: string;
  @Expose() updatedBy?: string;
}

File: src/backend/src/main/java/br/com/grupopipa/gestaointegrada/atendimento/agendamento/agendamento/dto/AgendamentoDTO.java
package br.com.grupopipa.gestaointegrada.atendimento.agendamento.agendamento.dto;

import java.time.LocalDateTime;
import java.util.List;
import java.util.UUID;

import com.fasterxml.jackson.annotation.JsonFormat;

import br.com.grupopipa.gestaointegrada.core.dto.DTO;
import lombok.Builder;
import lombok.Data;

@Data
@Builder
public class AgendamentoDTO implements DTO {

    private UUID id;
    private UUID agendaId;
    private String agendaNome;
    private UUID profissionalId;
    private String profissionalNome;
    private UUID pacienteId;
    private String pacienteNome;
    private UUID convenioId;
    private String convenioNome;
    private UUID categoriaId;
    private String categoriaNome;
    private UUID procedimentoId;
    private String procedimentoNome;
    private String observacao;
    private String status;
    private UUID atendimentoId;

    @JsonFormat(pattern = "yyyy-MM-dd'T'HH:mm")
    private List<LocalDateTime> horariosInicio;

    @JsonFormat(pattern = "yyyy-MM-dd'T'HH:mm")
    private List<LocalDateTime> horariosFim;

    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
    private String createdBy;
    private String updatedBy;
}
